src/bundle/Resources/richtext/stylesheets/xhtml5/edit/docbook.xsl

```diff
@@ -425,6 +440,13 @@
           <xsl:value-of select="@class"/>
         </xsl:attribute>
       </xsl:if>
+      <!-- Adding class and ezattribute from parent element (<figure>) -->
+      <xsl:if test="ancestor::ezxhtml5:figure">
+        <xsl:attribute name="class">
+          <xsl:value-of select="../@class"/>
+        </xsl:attribute>
+        <xsl:call-template name="ezattribute_from_figure"/>
+      </xsl:if>
```

Comment:
We could consider removing the line 496 and fix corresponding tests, as editor will no longer put `ezattribute` attributes on `<table>`:
```
<xsl:call-template name="ezattribute"/>
```

But this might be a considered BC break ?

```

Reply:
Maybe I am missing something, but given we don't risk malforming the output we should leave 496 line as-is. Wouldn't it be safer for people migrating to 4.5 from older versions? Other than that, the PR looks good to me.

Reply:
With Alloy we stored formatting for table like this in xhtml5 edit:
```<?xml version="1.0" encoding="UTF-8"?>
<section xmlns="http://ibexa.co/namespaces/ezpublish5/xhtml5/edit">
  <table class="tableClass" data-ezattribute-table-1="value 1" data-ezattribute-table-2="value 2" border="1" >
    <tbody>
...
```

With CKEditor, that is done like this:
```<?xml version="1.0" encoding="UTF-8"?>
<section xmlns="http://ibexa.co/namespaces/ezpublish5/xhtml5/edit">
  <figure class="tableClass" data-ezattribute-table-1="value 1" data-ezattribute-table-2="value 2" >
    <table>
      <tbody>
...
```

So, CKEditor does not put `class` and `data-ezattrobute*` attributes in `<table>` element. Therefore I ask if we should remove line 496(calling `ezattribute` template), and also line 438-442 ( check of `@class` on `<table>` )
This seems not to be needed by CKEditor. But we might keep it for BC ( People might still use XHTML Edit format when creating richtext content, so maybe we should not remove support for the mentioned attributes)

Point 2. My PR doesn't change richtext->xhtml edit. So it doesn't make a `<figure>` element when sending data to the editor. CKEditor seems to deal with that fine, it then automatically create the `<figure>` when it sends data back. I might add this `<figure>` element in xhtml edit if requested though.

point 3.
I investigated a bit further now.
 Alloy supported additional attributes on `table`, for instance :
```
  <table id="informaltable-id" class="tableClass" data-ezattribute-table-1="value 1" data-ezattribute-table-2="value 2" border="1" title="tableTitle" style="width:24px;border-width:42px;">
```

I don't get `id`, `border`, `title` or `style` to work with CKEditor, neither if I put them on `<figure>` or `<table`>. I would need @dew326 or others to comment on whatever that is possible to fix or not. At the moment we have a BC break in this regard, when comparing to 3.3

Reply:
@alongosz ^^^

Reply:
@vidarl it would keep it as-is, this might introduce a break hard to spot now. 

> Point 2. My PR doesn't change richtext->xhtml edit. So it doesn't make a <figure> element when sending data to the editor. CKEditor seems to deal with that fine, it then automatically create the <figure> when it sends data back. I might add this <figure> element in xhtml edit if requested though.

We might do it in the future if that element exists in DocBook, but for now for that bug fix, if it's possible, I'd avoid it.

> I don't get id, border, title or style to work with CKEditor, neither if I put them on <figure> or <table>. I would need @dew326 or others to comment on whatever that is possible to fix or not. At the moment we have a BC break in this regard, when comparing to 3.3

But is it just on the front-end level when OE creates payload or is it not stored in DocBook?

Reply:
> But is it just on the front-end level when OE creates payload or is it not stored in DocBook?

What I mean is that OE is not able to set the mentioned attributes on either `<figure>` or `<table>`  when it creates the payload, yes

Reply:
> > But is it just on the front-end level when OE creates payload or is it not stored in DocBook?
> 
> What I mean is that OE is not able to set the mentioned attributes on either `<figure>` or `<table>` when it creates the payload, yes

Hmm, then indeed we need @dew326's input once he's back from vacation. Merging this one for now, we can get back to it as a follow-up.